Removing empty dir on windows seems to file for project

Test error 

````
[smoke] > quarto render docs\project\site -o - => ./test.ts:121:72
error: AssertionError: Failed assertion:

--------------------------------------------------------------------------------
[smoke] > quarto render docs\project\site -o -
          run-tests.psl ..\..\..\..\..\..\C:\Users\chris\Documents\DEV_R\quarto-cli\tests\smoke\project\project-stdout.test.ts

[verify] > unknown

La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte. (os error 123), stat 'C:\Users\chris\Documents\DEV_R\quarto-cli\tests\docs\project\site\C:\Users\chris\AppData\Local\Temp\quarto-session96c8468e\9b1cbc71\dc6b9af2_files'
Error: La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte. (os error 123), stat 'C:\Users\chris\Documents\DEV_R\quarto-cli\tests\docs\project\site\C:\Users\chris\AppData\Local\Temp\quarto-session96c8468e\9b1cbc71\dc6b9af2_files'
    at Object.lstatSync (deno:runtime/js/30_fs.js:305:9)
    at existsSync (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/vendor/deno.land/std@0.166.0/fs/exists.ts:75:10)
    at removeIfEmptyDir (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/core/path.ts:47:7)
    at renderProject (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/command/render/project.ts:347:9)
    at async Command.fn (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/command/render/cmd.ts:193:26)
    at async Command.execute (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/vendor/deno.land/x/cliffy@v0.25.4/command/command.ts:1790:7)
    at async quarto (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/quarto.ts:111:3)
    at async Object.execute (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/tests/test.ts:68:5)
    at async fn (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/tests/test.ts:165:11)
    at async Object.testStepSanitizer [as fn] (deno:cli/js/40_testing.js:448:7)
````

That is definitely not correctly build 
````
C:\Users\chris\Documents\DEV_R\quarto-cli\tests\docs\project\site\C:\Users\chris\AppData\Local\Temp\quarto-session96c8468e\9b1cbc71\dc6b9af2_files
````

It comes from 
https://github.com/quarto-dev/quarto-cli/blob/9ce89221b1ac4f36a0826c8a9778b9e60b045f66/src/command/render/project.ts#L305-L312

If I had some log info 
````
      if (!keepFiles) {
        console.log(projDir);
        console.log(renderedFile.file);
        console.log(dirname(renderedFile.file));
        console.log(inputFilesDir(renderedFile.file));
````
this is the result
````
C:\Users\chris\Documents\DEV_R\quarto-cli\tests\docs\project\site
C:\Users\chris\AppData\Local\Temp\quarto-session96c8468e\9b1cbc71\dc6b9af2.html
C:\Users\chris\AppData\Local\Temp\quarto-session96c8468e\9b1cbc71
dc6b9af2_files
````

Definitely not right

--- a/news/changelog-1.4.md
+++ b/news/changelog-1.4.md
@@ -169,3 +169,4 @@
 - ([#6487](https://github.com/quarto-dev/quarto-cli/discussions/6487)): Fix `serviceworkers` check in `htmlDependency` to look at the correct key.
 - ([#6178](https://github.com/quarto-dev/quarto-cli/pull/6178)): When `QUARTO_LOG_LEVEL=DEBUG`, information about search for a R binary will be shown.
 - ([#5755](https://github.com/quarto-dev/quarto-cli/pull/5755)): Allow document metadata to control conditional content.
+- ([#6697](https://github.com/quarto-dev/quarto-cli/pull/6697)): Fix issue with outputing to stdout (`quarto render <file> -o -`) on Windows.

--- a/src/command/render/render.ts
+++ b/src/command/render/render.ts
@@ -13,7 +13,11 @@ import { Document, parseHtml } from "../../core/deno-dom.ts";
 import { mergeConfigs } from "../../core/config.ts";
 import { resourcePath } from "../../core/resources.ts";
 import { inputFilesDir } from "../../core/render.ts";
-import { normalizePath, pathWithForwardSlashes } from "../../core/path.ts";
+import {
+  normalizePath,
+  pathWithForwardSlashes,
+  safeExistsSync,
+} from "../../core/path.ts";
 
 import { FormatPandoc } from "../../config/types.ts";
 import {
@@ -469,7 +473,7 @@ export function renderResultFinalOutput(
 
   // if the final output doesn't exist then we must have been targetin stdout,
   // so return undefined
-  if (!existsSync(finalOutput)) {
+  if (!safeExistsSync(finalOutput)) {
     return undefined;
   }
 

--- a/tests/smoke/project/project-stdout.test.ts
+++ b/tests/smoke/project/project-stdout.test.ts
@@ -36,9 +36,6 @@ testQuartoCmd(
       if (existsSync(siteOutDir)) {
         await Deno.remove(siteOutDir, { recursive: true });
       }
-    },
-    // TODO: Make the test works for Windows
-    // https://github.com/quarto-dev/quarto-cli/issues/4194
-    ignore: Deno.build.os == "windows",
+    }
   },
 );

--- a/tests/smoke/render/render-stdout.test.ts
+++ b/tests/smoke/render/render-stdout.test.ts
@@ -1,0 +1,18 @@
+/*
+* project-render.test.ts
+*
+* Copyright (C) 2020-2023 Posit Software, PBC
+*
+*/
+import { testQuartoCmd } from "../../test.ts";
+import { docs } from "../../utils.ts";
+import { noErrorsOrWarnings } from "../../verify.ts";
+
+const input = docs("minimal.qmd");
+
+testQuartoCmd(
+  "render",
+  [input, "-o", "-"],
+  [noErrorsOrWarnings],
+  {},
+);

